lib/commons/math/has-visual-overlap.js

```diff
@@ -0,0 +1,44 @@
+import { visuallySort } from '../dom';
+
+/**
+ * Check if node A overlaps B
+ * @param {VirtualNode} vNodeA
+ * @param {VirtualNode} vNodeB
+ * @returns 0 | 1 | -1
+ */
+export default function hasVisualOverlap(vNodeA, vNodeB) {
+  const overlapRect = getRectOverlap(
```

Comment:
This could be a much simpler check (and easier to understand) if you used an [axis-aligned bounding box collision check](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/get-rect-stack.js#L129-L137). Then you just check for collision and call `visualSort` if they collide.

```js
return vNodeA.left < vNodeB.left + vNodeB.width &&
    vNodeA.left + vNodeA.width > vNodeB.left &&
    vNodeA.top < vNodeB.top + vNodeB.height &&
    vNodeA.top + vNodeA.height > vNodeB.top &&
    visuallySort(vNodeA, vNodeB) > 0;
```